python/map_closures/tools/visualizer.py

```diff
@@ -80,323 +109,482 @@ def update_closures(self, *kwargs):
-
-    def _update_closure(self, idx):
-        self.loop_closures_data.current_source.points = self.loop_closures_data.sources[idx]
-        self.loop_closures_data.current_source.transform(self.loop_closures_data.closure_poses[idx])
-        self.loop_closures_data.current_source.paint_uniform_color(RED)
-        self.loop_closures_data.current_target.points = self.loop_closures_data.targets[idx]
-        self.loop_closures_data.current_target.paint_uniform_color(GREEN)
+        reference_map.set_radius(self._states.reference_points_size, relative=False)
+        if self._states.align_closures:
+            reference_map.set_transform(query_map_pose @ self.loop_closures_data.closure_poses[idx])
+        else:
+            reference_map.set_transform(ref_map_pose)
+        reference_map.set_enabled(self._states.view_closure_reference)
 
-        self.closure_vis.update_geometry(self.loop_closures_data.current_source)
-        self.closure_vis.update_geometry(self.loop_closures_data.current_target)
+    def _background_color_callback(self):
+        changed, self._states.background_color = self._gui.ColorEdit3(
+            "Background Color",
+            self._states.background_color,
+        )
+        if changed:
+            self._ps.set_background_color(self._states.background_color)
+
+    def _global_view_callback(self):
+        button_name = LOCAL_VIEW_BUTTON if self._states.global_view else GLOBAL_VIEW_BUTTON
+        if self._gui.Button(button_name) or self._gui.IsKeyPressed(self._gui.ImGuiKey_V):
+            self._states.global_view = not self._states.global_view
+            if self._states.global_view:
+                self._ps.get_point_cloud("current_frame").set_transform(self._last_frame_pose)
+                self._ps.get_point_cloud("local_map").set_transform(
+                    self._last_frame_pose @ np.linalg.inv(self._last_frame_to_local_map_pose)
+                )
+                self._register_trajectory()
+            else:
+                self._ps.get_point_cloud("current_frame").set_transform(np.eye(4))
+                self._ps.get_point_cloud("local_map").set_transform(
+                    np.linalg.inv(self._last_frame_to_local_map_pose)
+                )
+                self._unregister_trajectory()
+            self._ps.reset_camera_to_home_view()
+
+    def _screenshot_callback(self):
+        posX = (
+            self._gui.GetCursorPosX()
+            + self._gui.GetColumnWidth()
+            - self._gui.CalcTextSize(SCREENSHOT_BUTTON)[0]
+            - self._gui.GetScrollX()
+            - self._gui.ImGuiStyleVar_ItemSpacing
+        )
+        self._gui.SetCursorPosX(posX)
+        if self._gui.Button(SCREENSHOT_BUTTON) or self._gui.IsKeyPressed(self._gui.ImGuiKey_S):
+            image_filename = "screenshot_" + (
+                datetime.datetime.now().strftime("%Y-%m-%d_%H-%M-%S") + ".jpg"
+            )
+            self._ps.screenshot(image_filename)
```

Comment:
I would suggest, since you have access to the result folder, to save there the screenshot. But it is just a suggestion, also the current directory is fine.

Reply:
This feature kind of feels unnecessary, as users can do a screenshot with own fancy tools.
Lets remove this completely.

Reply:
Yes let's remove it